packages/steps/src/BuildStepInput.ts

```diff
@@ -227,6 +212,19 @@ export class BuildStepInput<
     }
   }
 
+  private parseInputValueToString(value: string): string {
+    let parsedValue = value;
+    try {
+      parsedValue = JSON.parse(`"${value}"`);
+    } catch (err) {
+      // in case of SyntaxError return raw string
```

Comment:
When can it occur?

Reply:
It came up with manual testing after I fixed a failing test by also parsing the strings that get interpolated into the object properties, which fixed the test.
After your question I did some more testing and it turns out that it was the test that was broken and that additional parsing is not needed since the injected text is already parsed but the object property in which it gets injected is not and when the property itself is parsed later it causes the syntaxError.
I have removed the faulty test as well as additional parsing of the string injected into the object property since the property gets parsed later as a whole. Manual testing confirms this works